lib/puppet/util/posix.rb

```diff
@@ -1,5 +1,48 @@
 # Utility methods for interacting with POSIX objects; mostly user and group
 module Puppet::Util::POSIX
+  require 'ffi'
+  module GrpLibC
+    extend FFI::Library
+    ffi_lib FFI::Library::LIBC
+    begin
+      attach_function :getgrouplist, [:string, :int, :pointer, :pointer], :int
```

Comment:
I think `gid_t` is an unsigned int(?) if so then, the second parameter should be `:uint`